packages/angular_devkit/build_angular/src/differential-loading/is-differential-loading-needed.ts

```diff
@@ -0,0 +1,138 @@
+
+const es6ModuleSupport = require('./es6-module-support.json');
+const supportTable = convertSupportTableToBrowserVersionTuples(es6ModuleSupport);
+
+export function isDifferentialLoadingNeeded(
+  tsConfigFileName: string, query: string[] | undefined = undefined) {
+  const es5SupportNeeded = isEs5BrowserSupportNeeded(tsConfigFileName, query);
+  const target = getTargetVersion(tsConfigFileName);
+
+  return es5SupportNeeded && target !== ScriptTarget.ES5;
+}
+
+export function getTargetVersion(tsConfigFileName: string): ScriptTarget {
+  return _getTargetVersion(tsConfigFileName, new Set<string>());
+}
+
+function _getTargetVersion(tsConfigFileName: string, visited: Set<string>): ScriptTarget {
+  if (visited.has(tsConfigFileName)) {
+    throw new Error('cycle in tsconfig.json inheritance detected: '
+      + Array.from(visited).join(', ') );
+  }
+  visited.add(tsConfigFileName);
+
+  const content = require(tsConfigFileName);
+  const compilerOptions = content['compilerOptions'];
+  const target = compilerOptions['target'] as string;
+  const extendsField = content['extends'];
+
+  if (compilerOptions && target) {
+
+    const targetUC = target.toUpperCase();
+
+    // @ts-ignore
+    const scriptTarget = ScriptTarget[targetUC] as ScriptTarget;
+
+    if (typeof scriptTarget === 'undefined') {
+      throw new Error('Unsupported script target in tsconfig.json found: ' + target);
+    }
+
+    return scriptTarget;
+  } else if (extendsField) {
+    const tsConfigDirName = path.dirname(tsConfigFileName);
+    const nextTsConfigFileName = path.join(tsConfigDirName, extendsField);
+
+    return _getTargetVersion(nextTsConfigFileName, visited);
+  } else {
+    // Fallback to ES5
+    return ScriptTarget.ES5;
+  }
+
+}
+
+export function isEs5BrowserSupportNeeded(
+  tsConfigFileName: string | null, queries: string[] | undefined = undefined) {
+
+  const options: browserslist.Options = {};
+  if (tsConfigFileName) {
+    options.path = path.dirname(tsConfigFileName);
+  }
+
+  const browsersList: string[] = browserslist(queries, options);
+
+
+  return browsersList.some( (entry: string) => {
+    const [browserToSupport, versionToSupport] = parseBrowserVersionTuple(entry);
+
+    const minVersion = supportTable.get(browserToSupport);
+
+    if (typeof minVersion === 'undefined'
+      || cmpVersions(minVersion, versionToSupport) === 1) {
+      return true;
+    }
+
+    return false;
+  });
+}
+
+export function cmpVersions(version1: string, version2: string) {
+  const version1Tupel = toVersionTupel(version1);
+  const version2Tupel = toVersionTupel(version2);
+
+  if (version1Tupel[0] < version2Tupel[0]) {
+    return -1;
+  } else if (version1Tupel[0] > version2Tupel[0]) {
+    return 1;
+  } else if (version1Tupel[1] < version2Tupel[1]) {
+    return -1;
+  } else if (version1Tupel[1] > version2Tupel[1]) {
+    return 1;
+  } else {
+    return 0;
+  }
+}
+
+function toVersionTupel(version: string): [number, number] {
```

Comment:
Kindly confused what this tuples means. What is version `[0]` and what is `[1]`?

Reply:
It's about the major and the minor part of the version. And about comparing two versions with a major and a minor. I can replace the tuple with a class. 

Or do we already have such a function somewhere? 

In this case, reusing it would be the way to go, I guess ...

Reply:
The `semver` package can be used and is already a dependency.